core/src/test/java/io/grpc/internal/DnsNameResolverTest.java

```diff
@@ -888,14 +888,13 @@ public HttpConnectProxiedSocketAddress proxyFor(SocketAddress targetAddress) {
   }
 
   @Test
-  public void maybeChooseServiceConfig_failsOnMisspelling() {
+  public void maybeChooseServiceConfig_nullOnMispelling() {
```

Comment:
I think you should verify these all:
1. A `choice` with a bad key + a valid key, the choice is chosen(non-null).
2. A `choice` with only bad key, the choice is chosen.
3. A `choice` missing service config + bad key, throw. (this test case)
4. A `choice` missing service config + valid key, throw.

Also the test case name is not accurate.